Enhance: Add custom.js to Logseq publishing's app [minor behavior change]

candideu opened the pull request:
Re: https://github.com/logseq/logseq/issues/4425

I'm not a Clojure expert and haven't been able to test this, but I tried to search through the files and mimic what was done for custom.css and export.css. Please let me know if anything needs to be corrected.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/logseq/logseq?pullRequest=9291) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/logseq/logseq?pullRequest=9291) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/logseq/logseq?pullRequest=9291) it.</sub>

andelf commented on deps/publishing/src/logseq/publishing/html.cljs:
```diff
@@ -121,6 +121,7 @@ necessary db filtering"
         }
       }(window.location))"]
             ;; TODO: should make this configurable
+            [:script {:src "static/js/custom.js"}]
```
This leads to a missing file if custom.js is not configured.

candideu replied:
How can it be modified so that it's only added if there's a custom.js file?

For instance, export.css is added the same way in this file, but it's not automatically added to new Logseq graphs. It has to be manually added by the user (if needed), the same way that custom.js is added.

logseq-cldwalker replied:
@andelf This should be ok because the change ensures that custom.js is always written, in the same way that custom.css is

logseq-cldwalker commented:
@candideu Hi. This looks promising. This is a breaking change but I think it's reasonable to do as it matches custom.css's behavior. Could you fix the failing test and write a new test? For the new test, you can copy [create-export-with-css-files](reate-export-with-css-files), change custom.css to custom.js and remove export.css lines. If you want to run tests locally see https://github.com/logseq/logseq/tree/master/deps/publishing#testing

candideu commented:
> @candideu Hi. This looks promising. This is a breaking change but I think it's reasonable to do as it matches custom.css's behavior. Could you fix the failing test and write a new test? For the new test, you can copy [create-export-with-css-files](reate-export-with-css-files), change custom.css to custom.js and remove export.css lines. If you want to run tests locally see https://github.com/logseq/logseq/tree/master/deps/publishing#testing

Thank you! I wasn't able to open or find the create-export-with-css-files link. Could you send it again?

logseq-cldwalker commented:
Oops. Here it is
https://github.com/logseq/logseq/blob/016977059b8406412b4f5ed8a665abf0a6df3579/deps/publishing/test/logseq/publishing/export_test.cljs#L86-L98

On Thu, 4 May 2023 at 18:52, Candide U ***@***.***> wrote:

> @candideu <https://github.com/candideu> Hi. This looks promising. This is
> a breaking change but I think it's reasonable to do as it matches
> custom.css's behavior. Could you fix the failing test and write a new test?
> For the new test, you can copy create-export-with-css-files
> <http://reate-export-with-css-files>, change custom.css to custom.js and
> remove export.css lines. If you want to run tests locally see
> https://github.com/logseq/logseq/tree/master/deps/publishing#testing
>
> Thank you! I wasn't able to open or find the create-export-with-css-files
> link. Could you send it again?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/logseq/logseq/pull/9291#issuecomment-1535503445>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXFVC5ZGHNYU4OA6D6YBI5LXEQXMDANCNFSM6AAAAAAXSQOTJI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

candideu commented:
> Oops. Here it is https://github.com/logseq/logseq/blob/016977059b8406412b4f5ed8a665abf0a6df3579/deps/publishing/test/logseq/publishing/export_test.cljs#L86-L98

Thanks! I've added the test and updated my PR. Looks like the tests passed on GitHub. Let me know if I'm missing anything else.

logseq-cldwalker commented on deps/publishing/src/logseq/publishing/html.cljs:
```diff
@@ -121,6 +121,7 @@ necessary db filtering"
         }
       }(window.location))"]
             ;; TODO: should make this configurable
```
Missed this last review but app should load first

candideu replied:
Thanks, is there anything you need me to do @logseq-cldwalker ?

logseq-cldwalker replied:
nope but thanks for asking

logseq-cldwalker reviewed (approved):
@candideu Thanks for adding this support! 👍 🎋 Making a small change about when custom.js is loaded and this'll be good to go
